internal/wrphandlers/mocktr181/handler.go

```diff
@@ -115,53 +115,52 @@ func (h Handler) Enabled() bool {
 
 // HandleWrp is called to process a tr181 command
 func (h Handler) HandleWrp(msg wrp.Message) error {
-	payload := new(Tr181Payload)
-
-	err := json.Unmarshal(msg.Payload, &payload)
+	statusCode, payloadResponse, err := h.proccessCommand(msg.Payload)
 	if err != nil {
 		return errors.Join(err, wrpkit.ErrNotHandled)
 	}
 
-	var payloadResponse []byte
-	var statusCode int64
-
-	command := payload.Command
-
-	switch command {
-	case "GET":
-		statusCode, payloadResponse, err = h.get(payload)
-		if err != nil {
-			return errors.Join(err, wrpkit.ErrNotHandled)
-		}
-
-	case "SET":
-		statusCode, payloadResponse, err = h.set(payload)
-		if err != nil {
-			return errors.Join(err, wrpkit.ErrNotHandled)
-		}
-
-	default:
-		// currently only get and set are implemented for existing mocktr181
-		statusCode = 520
-		payloadResponse = []byte(fmt.Sprintf(`{"message": "command '%s' is not supported", "statusCode": %d}`, command, statusCode))
-	}
-
 	response := msg
 	response.Destination = msg.Source
 	response.Source = h.source
 	response.ContentType = "application/json"
 	response.Payload = payloadResponse
-
 	response.Status = &statusCode
-
-	err = h.egress.HandleWrp(response)
-	if err != nil {
+	if err = h.egress.HandleWrp(response); err != nil {
 		return errors.Join(err, wrpkit.ErrNotHandled)
 	}
 
 	return nil
 }
 
+func (h Handler) proccessCommand(wrpPayload []byte) (int64, []byte, error) {
+	var (
+		err             error
+		payloadResponse []byte
+		statusCode      = int64(520)
+	)
+
+	if len(wrpPayload) == 0 {
+		return statusCode, []byte(fmt.Sprintf(`{"message": ""Invalid Input Command"", "statusCode": %d}`, statusCode)), nil
```

Comment:
can we say empty so we know the payload is empty?

Reply:
I think we would but that's the expected behavior of the old c simulator.

we're trying to match the old c simulator as close as possible